### ISSUE TITLE
Duplicating Automations with Choose result in ghost/broken conditions

### ISSUE THREAD

**Issue by TheWebMachine** [home-assistant/frontend#16456]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

If you duplicate an Automation with the Choose function, editing the choices will leave ghosted option conditions in YAML that do not show in Visual mode, breaking the Automation. The extra conditions remain in YAML after deleting from visual editor.

![Screenshot_20230507_015606_Home Assistant](https://user-images.githubusercontent.com/1045052/236663602-18ec52a3-7262-49e2-b1be-e1f773a68781.jpg)
![Screenshot_20230507_015629_Home Assistant](https://user-images.githubusercontent.com/1045052/236663603-db75e056-3825-49c8-aa51-2c4bd4c2b04c.jpg)

### Describe the behavior you expected

When duplicating and then editing an Automation, YAML should always match Visual editor.

### Steps to reproduce the issue

1. Create an Automation that includes Choose in Actions
2. Duplicate the Automation and edit the Choose Conditions
3. Save and check YAML, which won't always match Visual editor and contains extra conditions on options that were deleted or edited in Visual

### What version of Home Assistant Core has the issue?

core-2023.5.2

### In which browser are you experiencing the issue with?

Home Assistant Android App beta-9972-8b32f9bf-full and Google Chrome v112

### Which operating system are you using to run this browser?

Android 12 and Ubuntu 22.04

### Javascript errors shown in your browser console/inspector

```txt
None
```

### Additional information

I recently switched to using Input Select lists for controlling scenes in my house for a cleaner interface (I have a lot of scenes). I created automations to monitor and change scenes based on when the Input Selects change. To save time, I duplicated the first automation since the structure was lengthy to recreate; merely editing the entities in Visual for each dup. Pretty standard. For days I couldn't figure out why some selections on some inputs just didn't activate the scene but the rest worked fine.

Trolling traces and logbook, I noticed the Automation YAML look wrong. Further inspection revealed that while Visual editor looked right, the actual YAML included extra conditions in those broken options beyond what Visual showed. They were remnants of the first Automation I duplicated from that didn't delete/edit when I first changed them in Visual.

Once I manually cleaned up the YAML to remove the extra conditions, which effected no change in Visual mind you, all of my options worked correctly and as expected. 

As an aside, it's interesting to note that I didn't know Choose Option Conditions were AND'd not OR'd, which is in contrast to Automation (edit: misstated "Conditions") Triggers being OR'd. If Choose Option Conditions were OR'd, I'd have never even noticed this Visual editor bug. 🤷🏻‍♂️  So, I learned something new there, too. 😎

**Comment by karwosts** [comment]
> which is in contrast to Automation Conditions being OR'd.

Automation conditions *are* AND, not OR. 

Anyway I tried your reproduce sequence with a trivial example and I didn't see anything wrong. 

Might suggest you post your exact yaml that you had a problem with.

**Comment by TheWebMachine** [comment]
> > which is in contrast to Automation Conditions being OR'd.
> 
> Automation conditions _are_ AND, not OR.
> 

I misspoke. I meant Automation Triggers, not Conditions. I've updated my previous comment to reflect the correction.

I have 10-15 options in use in each of the Automations and reproduction of this issue, even on my end, is very hit or miss. Only about 20% of the time the duplicate will have extra Conditions in the YAML not reflected in visual that are leftover from the original Automation that was duplicated. If I just keep spinning out dups, it eventually glitches one like in the screenshots above.

I noticed a bug and reported it in case anyone else noticed the same thing, as I only have my single HA instance to test with. If it doesn't meet the threshold to look into or can't be reliably reproduced by others, so be it. With so many moving parts, tracking down such random glitches isn't going to be easy or rewarding, obviously.
